### PR TITLE
Remove `woocommerce/experimental` dependency from `woocommerce/components`

### PR DESCRIPTION
**Pull request by louwie17**
This is a run off and (pre-cursor I guess) of #6923 where I need to make uses of the EllipsisComponent within the experimental package.
This currently causes a circular dependency, and this PR removes the dependencies we did have (only the `Text` component).

No changelog as it is in the components package.

## Test instructions

- Load this branch
- Do a quick smoke test of the home screen and the advanced filters within the analytics pages (**Analytics > Orders**)
- Generate the storybook (`npm run storybook`) and see if the experimental package is part of it now with the List component.
- Go to the List component under `components` and click on `docs`
- The subtitle `[depreciated] and will be replaced by ExperimentalList` should correctly link to the experimental list story.

### CODE REVIEW

File: packages/components/src/list/stories/index.js [hunk 1]
```diff
@@ -4,10 +4,15 @@
 import Gridicon from 'gridicons';
 import { withConsole } from '@storybook/addon-console';
 import {
-	List as ExperimentalList,
-	ListItem as ExperimentalListItem,
-	CollapsibleList as ExperimentalCollapsibleList,
-} from '@woocommerce/experimental';
```
Comment (louwie17) [inline packages/components/src/list/stories/index.js:10]: Moved the experimental story to the experimental package.

File: packages/components/src/list/stories/index.js [hunk 2]
```diff
@@ -35,7 +40,32 @@ function logItemClick( event ) {
...
+					<Title />
+					<Subtitle />
+					<Description
+						markdown={ `[depreciated] and will be replaced by
+						<a
+							data-sb-kind="woocommerce-admin-experimental-list"
+							data-sb-story="default"
+						>
+							ExperimentalList
+						</a>` }
```
Comment (louwie17) [inline packages/components/src/list/stories/index.js:60]: Add this for the subtitle

File: packages/components/src/list/stories/index.js [hunk 3]
```diff
@@ -35,7 +40,32 @@ function logItemClick( event ) {
...
+		withLinks,
+	],
+	parameters: {
+		docs: {
+			page: () => (
+				<>
+					<Title />
+					<Subtitle />
+					<Description
+						markdown={ `[depreciated] and will be replaced by
```
Comment (samueljseay) [inline packages/components/src/list/stories/index.js:54]: Should be *deprecated

**Review by samueljseay — DISMISSED** [review]
Looks good just a minor typo

File: packages/components/src/list/stories/index.js (hunk 3 above)
Comment (louwie17) [reply packages/components/src/list/stories/index.js]: Thanks should be fixed in https://github.com/woocommerce/woocommerce-admin/pull/6986/commits/0d715559ddd3eda1e3079d778af2a6112cc6a5e8

**Review by jeffstieler — APPROVED** [review]
👍

**Pull request merged**